### ISSUE TITLE
Rewrite deps in BFS order

### ISSUE THREAD

**Issue by Stebalien** [whyrusleeping/gx-go#50]
#47 fixed the *direct* issue in #44 but didn't fix the recursive case. In general, we should probably be rewriting BFS order (and never overwriting). That is, process a packages direct dependencies first before falling back on transitive dependencies, recursively.

**Comment by schomatis** [comment]
Agreed, I'll take this one, it seems like an excellent test case to see if I could apply the [DAG walker](https://github.com/ipfs/go-ipld-format/pull/39) here.

**Comment by schomatis** [comment]
Oh, unless this losses relevance with the lockfile support in https://github.com/whyrusleeping/gx-go/pull/49.
